.github/workflows/ci.yaml

```diff
@@ -12,70 +12,49 @@ on:
 
 jobs:
   gh-pages:
-    runs-on: macos-13
+    runs-on: macos-14
     needs: ["app"]
     steps:
-    # The default Java on macos-13 is 17. So we no longer need to set it.
-    # https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#java
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - run: make docs
     - name: Deploy docs to gh-pages
       uses: peaceiris/actions-gh-pages@v3
-      if: ${{ github.ref == 'refs/heads/main' }}
+      if: ${{ github.repository == 'authgear/authgear-sdk-android' && github.ref == 'refs/heads/main' }}
       with:
         github_token: ${{ secrets.GITHUB_TOKEN }}
         publish_dir: ./build/dokka/htmlMultiModule
+
   test:
-    runs-on: macos-13
+    runs-on: macos-14
     steps:
-    # The default Java on macos-13 is 17. So we no longer need to set it.
-    # https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#java
-    - uses: actions/checkout@v3
-    - name: Build sdk
-      run: ./gradlew sdk:assembleRelease
-    - name: Build sdk-okhttp
-      run: ./gradlew sdk-okhttp:assembleRelease
+    - uses: actions/checkout@v4
+    - run: make sdk
+    - run: make sdk-okhttp
+
   app:
```

Comment:
I think prs should not run this job?

Reply:
Not harmful to run it as long as we only upload in the `main` branch.

Reply:
But I think now all pr branches are running the upload job?

Reply:
Yes, you're right. I added a commit to fix this. But this PR is not ready yet. I still cannot figure out the permission issue to upload to Play Store.

Reply:
What permission issue?
https://docs.fastlane.tools/actions/upload_to_play_store/#setup
Does point 4 solves the problem? I think only Owner of the play store account can grant permission.